Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
+      bytes[3] == 'G') {
+    return _CGImageLoadPNG(bytes, length);
+  }
+
+  return _CGImageLoadImageWithWICDecoder(GUID_NULL, data, length);
+}
+
+CGImageRef _CGImageLoadGIF(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatGif, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadBMP(void *bytes, size_t length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadTIFF(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadPNG(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadJPEG(void *bytes, int length) {
+  return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes,
+                                         length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void *bytes,
+                                           int length) {
+  ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+  ComPtr<IWICBitmapDecoder> pDecoder;
+
+  if (decoderCls != GUID_NULL) {
+    ComPtr<IStream> spStream;
+    RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
+
+    ULONG written = 0;
+    HRESULT hr;
+
+    hr = spStream->Write(bytes, length, &written);
+    if (FAILED(hr) || (written != length)) {
+      TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr,
+                 length, written);
+      return NULL;
+    }
+
+    // seek to start
+    LARGE_INTEGER offset;
+    offset.QuadPart = 0;
+    RETURN_NULL_IF_FAILED(spStream->Seek(offset, STREAM_SEEK_SET, NULL));
+
+    RETURN_NULL_IF_FAILED(
+        imageFactory->CreateDecoder(decoderCls, NULL, &pDecoder));
+
+    hr = pDecoder->Initialize(spStream.Get(), WICDecodeMetadataCacheOnLoad);
+    RETURN_NULL_IF_FAILED(hr);
+
+  } else {
+    ComPtr<IWICStream> spStream;
+    RETURN_NULL_IF_FAILED(imageFactory->CreateStream(&spStream));
+
+    RETURN_NULL_IF_FAILED(spStream->InitializeFromMemory(
+        static_cast<unsigned char *>(bytes), length));
+    RETURN_NULL_IF_FAILED(imageFactory->CreateDecoderFromStream(
+        spStream.Get(), nullptr, WICDecodeMetadataCacheOnDemand, &pDecoder));
+  }
+
+  ComPtr<IWICBitmapFrameDecode> bitMapFrameDecoder;
+  RETURN_NULL_IF_FAILED(pDecoder->GetFrame(0, &bitMapFrameDecoder));
+
+  CGImageRef imageRef = __CGImage::CreateInstance();
+  imageRef->ImageSource().Attach(bitMapFrameDecoder.Detach());
+  // TODO:(JJ) Set the color space
+  return imageRef;
+}
+
+NSData *_CGImagePNGRepresentation(CGImageRef image) {
+  return _CGImageRepresentation(image, GUID_WICPixelFormat32bppBGRA);
+}
+
+NSData *_CGImageJPEGRepresentation(CGImageRef image) {
```

Comment:
>_CGImageJPEGRepresentation [](start = 8, length = 26)

It doesn't look like this actually wraps it in a container format ... it just returns the bytes after converting them to another pixel format.